plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureStorageCleanupThirdPartyTests.java

```diff
@@ -71,5 +81,32 @@ protected void createRepository(String repoName) {
                 .put("base_path", System.getProperty("test.azure.base"))
             ).get();
         assertThat(putRepositoryResponse.isAcknowledged(), equalTo(true));
+        if (Strings.hasText(System.getProperty("test.azure.sas_token"))) {
+            ensureSasTokenHasMinimalPermissions();
+        }
+    }
+
+    private void ensureSasTokenHasMinimalPermissions() {
```

Comment:
I'd call this `ensureSasTokenPersmissions()` and maybe also test other operations like listing all root blobs or delete the container?

Reply:
@tlrx so sorry for the delay here, somehow I missed this one!

+1 on the rename, the method name is weird and the assertion text documents what's going on here anyway.

> listing all root blobs

No sure that's a thing. I think you can only set up the list permission per container right, you can't define permissions per folder inside a container can you?

> delete the container

Hmm, I would argue here that using the wrong key and actually running into this actually deleting the container has some really annoying ramifications and I can't see us ever adding code that deletes a container?

-> I would argue this level of testing might be good enough for the intended purpose. In the end this change is mainly about documenting the need for restricted permissions for these tokens on CI. "Others" will run tests against Azure with production kind tokens as well anyway so this is mainly about our CI not accidentally switching to some default-all-permissions-enabled token and us running into the issues we've been running into as a result. I'd see this whole change more as a documentation change :)